src/MIDebugPackage/OpenFolderSchema.json

```diff
@@ -0,0 +1,224 @@
+{
+  "debugExtensions": {
+    "cppdbg": {
+      "vsDebugEngineGuid": "{ea6637c6-17df-45b5-a183-0951c54243bc}",
+      "VsDebugTargetInfo4": {
+        "bstrCurDir": "${workspaceRoot}",
+        "bstrExe": "${debugInfo.target}",
+        "bstrOptions": "*"
+      },
+      "templates": [
+        {
+          "templateId": "launch",
+          "displayName": "C/C++ Launch for MinGW/Cygwin (gdb)",
+          "description": "Debug launch configuration for C/C++ hosted in MinGW or Cygwin.",
+          "fileExtensions": [ "*.exe", "makefile" ],
+          "initialConfiguration": {
+            "cwd": "${workspaceRoot}",
+            "program": "${debugInfo.target}",
+            "MIMode": "gdb",
+            "miDebuggerPath": "${env.MINGW_PREFIX}\\bin\\gdb.exe",
+            "externalConsole": true
+          }
+        },
+        {
+          "templateId": "attach",
+          "displayName": "C/C++ Attach for MinGW/Cygwin (gdb)",
+          "description": "Debug attach configuration for C/C++ hosted in MinGW or Cygwin.",
+          "fileExtensions": [ "*.exe", "makefile" ],
+          "initialConfiguration": {
+            "cwd": "${workspaceRoot}",
+            "program": "${debugInfo.target}",
+            "processId": 0,
+            "MIMode": "gdb",
+            "miDebuggerPath": "${env.MINGW_PREFIX}\\bin\\gdb.exe",
+            "externalConsole": true
```

Comment:
nit: `externalConsole` doesn't do anything for attach, so we might not want to have it in the template in the name of reducing its size.